liblouis/lou_backTranslateString.c

```diff
@@ -1534,7 +1534,8 @@ back_passDoAction(const TranslationTableHeader *table, int *pos, int mode,
 	int newPos = match.endReplace;
 
 	if (!copyCharacters(match.startMatch, match.startReplace, table, mode, input, output,
-				posMapping, cursorPosition, cursorStatus, ctx, currentOpcode))
+				posMapping, cursorPosition, cursorStatus, ctx, currentOpcode) ||
+			match.startReplace < 0)
```

Comment:
I would put `if (match.startReplace < 0) return 0` on a seperate line above, or at least move `match.startReplace < 0` before `!copyCharacters(...)`.

But what I would prefer even more is understanding why `match.startReplace` would ever be smaller than 0. When `back_passDoAction` should normally only be called when `back_passDoTest` returned `1`, and `back_passDoTest` should not return `1` when `match->startReplace` was not set.

Reply:
> When `back_passDoAction` should normally only be called when `back_passDoTest`

The `back_passDoAction` called by `makeCorrections` function in the #1719 is called when `currentOpcode` is `CTO_Correct`, which is not related to the `back_passDoTest ` to me?

> When `back_passDoAction` should normally only be called when `back_passDoTest` returned `1`, and `back_passDoTest` should not return `1` when `match->startReplace` was not set.

`match->startReplace` is set when `(*passInstructions)[*passIC]` is 91 or 32, and I insert some print statement:

```
back_passDoTest: (*passInstructions)[*passIC] = 33
back_passDoTest: (*passInstructions)[*passIC] = 34
findBackPassRule return 1
Out while: match.startReplace = -1
Before back_passDoAction: match.startReplace = -1
```

Reply:
Yes, it is initialized as `-1` and not set in the following logic (i.e., not in the 91 or 32 switch case).